pkg/controllers/oauthclientscontroller/oauthclientscontroller.go

```diff
@@ -158,16 +158,24 @@ func randomBits(bits uint) []byte {
 	return b
 }
 
-func ensureOAuthClient(ctx context.Context, oauthClients oauthclient.OAuthClientInterface, client oauthv1.OAuthClient) error {
-	_, err := oauthClients.Create(ctx, &client, metav1.CreateOptions{})
-	if err == nil || !apierrors.IsAlreadyExists(err) {
+func ensureOAuthClient(ctx context.Context, oauthClients oauthclient.OAuthClientInterface, oauthClientLister oauthv1listers.OAuthClientLister, client oauthv1.OAuthClient) error {
```

Comment:
@kramaranya @ibihim what do you think about making this a method on `oauthsClientsController`, instead of passing in the client and the lister?

Reply:
Functions are nearly always better than methods, but if they get like plenty of arguments from the object... it might be better to make them methods :)